go-controller/pkg/util/go_ovn.go

```diff
@@ -0,0 +1,776 @@
+package util
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"syscall"
+	"time"
+
+	goovn "github.com/ebay/go-ovn"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"k8s.io/klog"
+)
+
+type OVNClient struct {
+	client goovn.Client
+}
+
+var _ OVNInterface = &OVNClient{}
+
+// these are the interface objects corresponding to
+// nb client and sb client respectively.
+var OVNNBDBClient OVNInterface
+var OVNSBDBClient OVNInterface
+
+func InitOVNDBClients() error {
```

Comment:
so you implemented these clients as global instance, rather than an instance per init returned to the caller. Would it be better to do the latter and allow the OVN controller for example to hold its own client. So that if needed in the future there could be multiple client connections?

Reply:
I think it would make sense but I have to check if there are places where we call into go-ovn bindings without a controller reference. We would have to pass the controller/client reference around. 

That being said, the client instances cache all the nbdb/sbdb rows so it is definitely a memory hit if we decide we need multiple clients. The client code on the plus side is go-routine safe. So we have to carefully weigh the need for multiple client instances per work-queue in the future.

Reply:
FWIW, I was following the model with the Nbctl/Sbctl where the commands are invoked directly from the util package.

Reply:
Yeah... that is not the model to follow :-). Generally speaking, global clients are to be avoided.

Reply:
I see the point of avoiding global clients and embedding the instance in the controller itself for sake of encapsulation and to follow good software engineering principles. 

But if we are never going to create more than one controller and not going to have more than one instance of the ovndb clients (one per database - northbound and southbound) is it really fundamentally any different?

Reply:
Made the ovn clients instances in the controller.